openml/datasets/functions.py

```diff
@@ -171,44 +182,44 @@ def check_datasets_active(dids):
     Parameters
     ----------
     dids : iterable
-        A list of integers representing dataset ids.
+        Integers representing dataset ids.
 
     Returns
-    active : dict of int to boolean
+    -------
     dict
-        A dictionary with items {did: active}, where active is a boolean. It
-        is set to True if the dataset is active.
+        A dictionary with items {did: bool}
     """
     dataset_list = list_datasets()
     dids = sorted(dids)
     active = {}
 
-    dataset_list_idx = 0
     for did in dids:
-        # TODO replace with a more efficient while loop!
-        for idx in range(dataset_list_idx, len(dataset_list)):
-            if did == dataset_list[idx]['did']:
-                active['did'] = bool(dataset_list[idx]['status'])
-        dataset_list_idx = idx
+        for dataset in dataset_list:
```

Comment:
Maybe that tells us that ``list_datasets`` should actually return a dict indexed by ``did``?